### ISSUE TITLE
"Recursion detected" error prevents Go-To's from eventually referencing themselves

### ISSUE THREAD

**Issue by Crystalwarrior** [AnidemDex/Blockflow#114]
![image](https://github.com/AnidemDex/Blockflow/assets/3470436/3c0b63c1-8626-4614-a427-eab2bdb26ded)
This means you can't set up back-and-forth goto's, each goto must result in a new timeline.

**Comment by AnidemDex** [comment]
There is a way to make cross-referencing resources in godot 4 using resource UID, maybe we can use that? Path is also a good solution

**Comment by Crystalwarrior** [comment]
UID is actually a great option if it's consistent cuz with paths you might have an issue if the project is reorganized

**Comment by AnidemDex** [comment]
The only problem with using UID is that _maybe_ we can break current stored resource paths in user projects :thinking:

**Comment by Crystalwarrior** [comment]
Right, maybe we can add an auto converter in some way? I'll do some brainstorming. Do we save a version number with the resource?

**Comment by AnidemDex** [comment]
We still don't save any metadata which identifies the resource format yet, because we haven't defined one, so godot handles the data
